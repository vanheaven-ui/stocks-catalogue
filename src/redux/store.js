import { compose, createStore } from 'redux';
import rootReducer from './reducers';

const storeEnhancers = compose(
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), // eslint-disable-line
);

const store = createStore(rootReducer, storeEnhancers);

export default store;
