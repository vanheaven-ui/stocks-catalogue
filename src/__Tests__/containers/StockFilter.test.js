import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import StocksFilter from '../../containers/StockFilter';
import store from '../../redux/store';
import rootNode from '../helpers';

describe('StockFilter Container component', () => {
  test('renders without crashing', () => {
    ReactDOM.render(<Provider store={store}><StocksFilter /></Provider>, rootNode());
  });

  test('doesnot change unexpectedly', () => {
    const tree = renderer.create(<Provider store={store}><StocksFilter /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
