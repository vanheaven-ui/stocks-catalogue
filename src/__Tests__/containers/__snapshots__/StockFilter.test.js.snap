// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockFilter Container component doesnot change unexpectedly 1`] = `
<section
  className="filters"
  style={
    Object {
      "backgroundColor": "#ffcd0a",
      "color": "#d1450d",
    }
  }
>
  <h2>
    Browse Stocks with Filters:
  </h2>
  <div
    className="container"
  >
    <form
      onSubmit={[Function]}
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="filters"
      >
        <select
          name="PRICE"
          onChange={[Function]}
          value="All"
        >
          <option
            value="default"
          >
            Select:
             
            PRICE RANGE
          </option>
          <option
            value="<100"
          >
            &lt;100
          </option>
          <option
            value="100 < Price < 200"
          >
            100 &lt; Price &lt; 200
          </option>
          <option
            value="200 < Price < 300"
          >
            200 &lt; Price &lt; 300
          </option>
          <option
            value="300 < Price < 400"
          >
            300 &lt; Price &lt; 400
          </option>
          <option
            value="400 < Price < 500"
          >
            400 &lt; Price &lt; 500
          </option>
          <option
            value=">500"
          >
            &gt;500
          </option>
        </select>
        <select
          name="EXCHANGE"
          onChange={[Function]}
          value="All"
        >
          <option
            value="default"
          >
            Select:
             
            EXCHANGE
          </option>
        </select>
        <button
          type="submit"
        >
          Apply
        </button>
      </div>
    </form>
  </div>
</section>
`;
